packager/signer/signer.go

```diff
@@ -219,6 +219,62 @@ func (this *Signer) fetchURL(fetch *url.URL, serveHTTPReq *http.Request) (*http.
 	return req, resp, nil
 }
 
+// Some Content-Security-Policy (CSP) configurations have the ability to break
+// AMPHTML document functionality on the AMPHTML Cache if set on the document.
+// This method parses the publisher's provided CSP and mutates it to ensure
+// that the document is not broken on the AMP Cache.
+//
+// Specifically, the following CSP directives are passed through unmodified:
+//  - base-uri
+//  - block-all-mixed-content
+//  - font-src
+//  - form-action
+//  - manifest-src
+//  - referrer
+//  - upgrade-insecure-requests
+// And the following CSP directives are overridden to specific values:
+//  - object-src
+//  - report-uri
+//  - script-src
+//  - style-src
+//  - default-src
+// All other CSP directives are stripped from the publisher provided CSP.
+func MutateFetchedContentSecurityPolicy(fetched string) (string) {
+	directiveTokens := strings.Split(fetched, ";")
+	var newCsp string
+	for i := range directiveTokens {
+		directiveToken := directiveTokens[i]
+		trimmed := strings.TrimSpace(directiveToken)
+		directiveParts := strings.Fields(trimmed)
```

Comment:
Add a note that this differs slightly from the spec in that it includes `U+000b vertical tab` in its definition of white space.

(Sources: https://infra.spec.whatwg.org/#ascii-whitespace vs https://golang.org/pkg/strings/#Fields, https://golang.org/pkg/unicode/#IsSpace, and http://xahlee.info/golang/golang_string_backslash_escape.html)